### PR TITLE
Set Providers Info

### PR DESCRIPTION
**Pull request by hjamaly**
This PR introduces two new screens for the provider sign-up process, integrates Firebase for storing relevant data, and adds a compact test to simulate provider registration.

**Changes Implemented:**

_First Screen:_

Enables providers to input service-related details:

- Service offered
- Short description
- Minimum price
- Languages spoken
- Profile picture (uploaded from the gallery).

Information is validated and stored in Firebase under the provider's profile.

_Second Screen:_

Proposes the option to create service packages:
Context: As discussed in previous stand-ups, packages are optional as they depend on the seeker's specific needs.
If the provider opts to create packages, they can add up to three packages with:

- Package title
- Price
- Description
- Key features (bullet points).

Packages are stored in a separate Firebase collection with the provider ID as a field for filtering and linkage.

_Firebase Integration:_
Updated Firebase structure:

- Provider profile includes fields for service details.
- A separate collection for packages, linked to the provider via the providerId field for filtering.

**Testing:**

Added a compact test to simulate a complete provider registration:

- Covers filling in service information and (optionally) creating packages.

**Verification Steps:**

Create a provider profile using the new sign-up screens.
Confirm that:

- Provider service details are stored in Firebase under the provider profile.
- Packages (if created) are correctly stored in the packages collection and linked to the provider via providerId.

### CODE REVIEW

**Review by manunam2002 — APPROVED** [review]
### feedback

This PR allows the provider to complete his profile with all the informations that were previously hardcoded. It is indeed a huge and valuable addition to the project and you did a great work!

#### Areas for improvements
- Please look at my comments in the relevant files

File: app/src/androidTest/java/com/android/solvit/EndToEndTestCreateProfile.kt
Comment (yshdb5) [inline app/src/androidTest/java/com/android/solvit/EndToEndTestCreateProfile.kt:1]: LGTM

File: app/src/main/java/com/android/solvit/shared/model/provider/ProviderRepositoryFirestore.kt [hunk 1]
```diff
@@ -152,4 +171,22 @@ class ProviderRepositoryFirestore(private val db: FirebaseFirestore) : ProviderR
       }
     }
   }
+
+  private fun uploadImageToStorage(
```
[the remark this answers is not in the document]
Comment (yshdb5) [reply app/src/main/java/com/android/solvit/shared/model/provider/ProviderRepositoryFirestore.kt:175]: +1

**Review by rimabk — APPROVED** [review]
**PR REVIEW:**
This is a good feature that enables the provider to complete their registration seamlessly by allowing them to select services, provide descriptions, upload images, and offer service packages.
**Strengths**:
**Clean and Structured Code:** The code is well-organized into clear sections, making it easy to follow and understand.
**Image Upload:** The `UploadImage` composable effectively handles image uploads and shows either the uploaded image or a placeholder icon. This is a valuable feature for any profile-based application.
**Suggestions for Improvement:**
**Package Validation:** Add validation for the required fields in the package section (name, price, details, features) before allowing the user to proceed. This would prevent incomplete package submissions.
**Enhanced Form Feedback:** Consider adding more granular error messages for all fields to improve user experience and guide them through the form filling process.
Overall, your code is robust and well-organized. Keep up the great work!

File: app/src/main/java/com/android/solvit/seeker/ui/provider/ProviderRegistrationProfile.kt [hunk 2]
```diff
@@ -311,8 +420,34 @@ fun ProviderRegistrationScreen(
...
+                              service = Services.valueOf(selectedService),
+                              description = description,
+                              price = startingPrice.toDouble(),
+                              languages = selectedLanguages.map { Language.valueOf(it) },
                               location = loc)
-                      viewModel.addProvider(newProviderProfile)
+                      // Check that the provider want to offer service packages
+                      Log.e("ServicePackages", "offerPackages : ${offerPackages.value}")
+                      if (offerPackages.value && packagesNames.isNotEmpty()) {
+                        for (i in 0..2) {
```
[the remark this answers is not in the document]
Comment (yshdb5) [reply app/src/main/java/com/android/solvit/seeker/ui/provider/ProviderRegistrationProfile.kt:431]: +1

File: app/src/main/java/com/android/solvit/seeker/ui/provider/ProviderRegistrationProfile.kt [hunk 3]
```diff
@@ -325,3 +460,335 @@ fun ProviderRegistrationScreen(
...
+            selectedImageUri = providerImageUri,
+            imageUrl = null,
+            onImageSelected = { uri -> onImageSelected(uri) })
+
+        // Enter Provider Brief description
+        CustomOutlinedTextField(
+            value = description,
+            onValueChange = { onDescriptionChange(it) },
+            label = "About you",
+            placeholder =
```
Comment (yshdb5) [inline app/src/main/java/com/android/solvit/seeker/ui/provider/ProviderRegistrationProfile.kt:526]: There is no limit for the description size. I don't know if it should be taken care of here or where the description is used but I was just thinking about it.

Comment (yshdb5) [reply app/src/main/java/com/android/solvit/seeker/ui/provider/ProviderRegistrationProfile.kt:526]: The rest LGTM!

**Review by yshdb5 — APPROVED** [review]
### Overall Comment

This pull request enhances the Provider's registration process and the package proposal feature by updating the `ProviderRegistrationProfile.kt` file, the `PackageProposal` model, integrating `PackageProposalRepositoryFirestore`, and modifying tests and other UI components.

### Strong Elements

1. **Comprehensive Updates**: The PR addresses models, UI, and tests, indicating a thorough enhancement approach.
2. **Enhanced Data Models**: Adding `packageNumber` and `providerId` to `PackageProposal` enriches data management and usability of this class.
3. **UI Enhancements**: New UI elements for package proposals and provider registration improve user experience.

### Areas for Improvement
- I left comments for potential improvements in the files, overall it's a great job!

### Conclusion
This PR improves the application's functionality and user experience. Overall, excellent work on these improvements!

File: app/src/main/java/com/android/solvit/seeker/ui/provider/ProviderRegistrationProfile.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (hjamaly) [reply app/src/main/java/com/android/solvit/seeker/ui/provider/ProviderRegistrationProfile.kt:431]: I more thought to it for the UI with packages proposals ! I found it not consistent with only one package and also I think it's better for the seeker to offer him the possibility of choices ! We can re talk about it during next stand up

File: app/src/main/java/com/android/solvit/seeker/ui/provider/ProviderRegistrationProfile.kt (hunk 3 above)
Comment (hjamaly) [reply app/src/main/java/com/android/solvit/seeker/ui/provider/ProviderRegistrationProfile.kt:526]: Yes ! we can add a condition ! What about a max of 50 words ?

**Pull request merged**